src/main.js

```diff
@@ -48,6 +55,27 @@ angular.module(MODULE_NAME, [])
       },
 
       link(scope, element, attrs, ngModel) {
+        const _setViewValue = () => {
+          let viewValue;
+          switch (_dateFormat) {
```

Comment:
I see what you're saying here, @graingert, but changing `_setViewValue` to `getViewValue` in the way you're describing requires more code to be added to the two contexts in which `_setViewValue` is currently being called. Line 131 could be changed to:
```
scope.saveUpdateDate = function() {
  ngModel.$setViewValue(getViewValue());
}
```
Also on line 140
```
scope.save = function () {
  return saveFn(scope.$parent, { $value: _setViewValue() });
};
```
could be changed to 
```
scope.save = function () {
  const viewValue = getViewValue();
  ngModel.$setViewValue(viewValue);
  return saveFn(scope.$parent, { $value: viewValue });
};
```
but in both instances we're adding more code. I think it makes sense to have `ngModel.$setViewValue()` encapsulated in `_setViewValue()` the way it is now to keep the code simpler and DRYer.